src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -39,59 +33,92 @@ const propTypes = {
     A positive value shifts the tooltip down, and a negative value shifts it up. */
     shiftVertical: PropTypes.number.isRequired,
 
-    /** Callback to set the Ref to the Tooltip */
-    setTooltipRef: PropTypes.func.isRequired,
-
     /** Text to be shown in the tooltip */
     text: PropTypes.string.isRequired,
 
-    /** Callback to be used to calulate the width and height of tooltip */
-    measureTooltip: PropTypes.func.isRequired,
-};
+    /** Number of pixels to set max-width on tooltip  */
+    maxWidth: PropTypes.number.isRequired,
 
-const defaultProps = {};
-
-const TooltipRenderedOnPageBody = (props) => {
-    const {
-        animationStyle,
-        tooltipWrapperStyle,
-        tooltipTextStyle,
-        pointerWrapperStyle,
-        pointerStyle,
-    } = getTooltipStyles(
-        props.animation,
-        props.windowWidth,
-        props.xOffset,
-        props.yOffset,
-        props.wrapperWidth,
-        props.wrapperHeight,
-        props.tooltipWidth,
-        props.tooltipHeight,
-        props.shiftHorizontal,
-        props.shiftVertical,
-    );
-    return ReactDOM.createPortal(
-        <Animated.View
-            ref={props.setTooltipRef}
-            onLayout={props.measureTooltip}
-            style={[tooltipWrapperStyle, animationStyle]}
-        >
-            <Text style={tooltipTextStyle} numberOfLines={1}>{props.text}</Text>
-            <View style={pointerWrapperStyle}>
-                <View style={pointerStyle} />
-            </View>
-        </Animated.View>,
-        document.querySelector('body'),
-    );
+    /** Maximum number of lines to show in tooltip */
+    numberOfLines: PropTypes.number.isRequired,
 };
 
-TooltipRenderedOnPageBody.propTypes = propTypes;
-TooltipRenderedOnPageBody.defaultProps = defaultProps;
-TooltipRenderedOnPageBody.displayName = 'TooltipRenderedOnPageBody';
-
 // Props will change frequently.
 // On every tooltip hover, we update the position in state which will result in re-rendering.
 // We also update the state on layout changes which will be triggered often.
 // There will be n number of tooltip components in the page.
-// Its good to memorize this one.
-export default memo(TooltipRenderedOnPageBody);
+// It's good to memorize this one.
+class TooltipRenderedOnPageBody extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            // The width of toolttip's inner text
+            tooltipTextWidth: 0,
+
+            // The width and height of the tooltip itself
+            tooltipWidth: 0,
+            tooltipHeight: 0,
+        };
+
+        this.measureTooltip = this.measureTooltip.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({
+            tooltipTextWidth: this.textRef.offsetWidth,
+        });
+    }
+
+    /**
+     * Measure the size of the tooltip itself.
+     *
+     * @param {Object} nativeEvent
```

Comment:
NAB; this could be more clear - what kind of native event? And where does it come from?

Reply:
Sorry for delay.
It comes from `Animated.View`'s  `onLayout` prop. [`onLayout`](https://reactnative.dev/docs/view#onlayout) is prop that many react-native's core components have. It returns a event object wich has another object called 'nativeEvent' which is a type of  [LayoutEvent Object](https://reactnative.dev/docs/layoutevent). The LayoutEvent object has layout info (i.e. width, height) about the component where `onLayout` prop is being used,

https://github.com/Expensify/App/blob/89c578184ea9475e4f43531eb84e5eb42d2964f8/src/components/Tooltip/TooltipRenderedOnPageBody.js#L106-L109

Maybe we can make param comment: 

```js
* @param {Object} nativeEvent - LayoutEvent
```

Reply:
Aah thanks so much for explaining - on second thought I think it's fine as is, it is kinda self explanatory and a very small function :D